kura/org.eclipse.kura.driver.ble.xdk/src/main/java/org/eclipse/kura/internal/driver/ble/xdk/XdkDriver.java

```diff
@@ -0,0 +1,678 @@
+        private final String xdkAddress;
+        private final SensorName sensorName;
+        private final ChannelRecord channelRecord;
+
+        public XdkRequestInfo(final ChannelRecord channelRecord, final DataType dataType,
+                final String xdkAddress, final SensorName sensorName) {
+            this.dataType = dataType;
+            this.xdkAddress = xdkAddress;
+            this.sensorName = sensorName;
+            this.channelRecord = channelRecord;
+        }
+
+        private static void fail(final ChannelRecord record, final String message) {
+            record.setChannelStatus(new ChannelStatus(FAILURE, message, null));
+            record.setTimestamp(System.currentTimeMillis());
+        }
+
+        public static Optional<XdkRequestInfo> extract(final ChannelRecord record) {
+            final Map<String, Object> channelConfig = record.getChannelConfig();
+            final String xdkAddress;
+            final SensorName sensorName;
+
+            try {
+                xdkAddress = XdkChannelDescriptor.getXdkAddress(channelConfig);
+            } catch (final Exception e) {
+                fail(record, "Error while retrieving Xdk address");
+                logger.error("Error retrieving Xdk Address", e);
+                return Optional.empty();
+            }
+
+            try {
+                sensorName = XdkChannelDescriptor.getSensorName(channelConfig);
+            } catch (final Exception e) {
+                fail(record, "Error while retrieving sensor name");
+                logger.error("Error retrieving Sensor name", e);
+                return Optional.empty();
+            }
+
+            final DataType dataType = record.getValueType();
+
+            if (isNull(dataType)) {
+                fail(record, "Error while retrieving value type");
+                return Optional.empty();
+            }
+
+            return Optional.of(new XdkRequestInfo(record, dataType, xdkAddress, sensorName));
+        }
+    }
+
+	@Override
+	public PreparedRead prepareRead(List<ChannelRecord> channelRecords) {
+		requireNonNull(channelRecords, "Channel Record list cannot be null");
+
+        try (XdkPreparedRead preparedRead = new XdkPreparedRead()) {
+            preparedRead.channelRecords = channelRecords;
+
+            for (ChannelRecord record : channelRecords) {
+                XdkRequestInfo.extract(record).ifPresent(preparedRead.requestInfos::add);
+            }
+            return preparedRead;
+        }
+	}
+	 
+	 private void registerNotification(SensorListener sensorListener) {
+	        switch (sensorListener.getSensorType()) {
+	        case "ACCELERATION_X":
+	        	sensorListener.getXdk().setSamplingRate((int) (sensorListener.getPeriod() / 100));
+	        	sensorListener.getXdk().disableHighNotifications();
+	        	sensorListener.getXdk().enableHighNotifications(SensorListener.getSensorConsumer(sensorListener));
+
+	            break;
+	        case "ACCELERATION_Y":
```

Comment:
Since most of the switch cases share the same code, please consider merging them to reduce code duplication